projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-wpcom-offline-subscription-service.php

```diff
@@ -52,6 +53,57 @@ public function visitor_can_view_content( $valid_plan_ids, $access_level ) {
 		return $this->validate_subscriptions( $valid_plan_ids, $subscriptions );
 	}
 
+	/**
+	 * Check if the subscriber can receive the newsletter
+	 *
+	 * @param int $user_id User id.
+	 * @param int $blog_id Blog id.
+	 * @param int $post_id Post id.
+	 *
+	 * @return bool
+	 * @throws \Exception Throws an exception when used outside of WPCOM.
+	 */
+	public static function subscriber_can_receive_post_by_mail( $user_id, $blog_id, $post_id ) {
+
+		// Site admins can do everything
+		if ( current_user_can( 'edit_post', $post_id ) ) {
+			return true;
+		}
+
+		switch_to_blog( $blog_id );
+
+		$previous_user = wp_get_current_user();
+		wp_set_current_user( $user_id );
+
+		/**
+		 * Filter the subscriptions attached to a specific user on a given site.
+		 *
+		 * @since 9.4.0
+		 *
+		 * @param array $subscriptions Array of subscriptions.
+		 * @param int   $user_id The user's ID.
+		 * @param int   $site_id ID of the current site.
+		 */
+		$subscriptions = apply_filters(
+			'earn_get_user_subscriptions_for_site_id',
+			array(),
+			wp_get_current_user()->ID,
+			$blog_id
+		);
+
+		$access_level       = get_post_meta( $post_id, '_jetpack_newsletter_access', true );
```

Comment:
Would be sweet to break it out as well into a method

Reply:
Now that I reused `visitor_can_view_content()` ( see https://github.com/Automattic/jetpack/pull/27352/files#r1021249651), I am not sure it is still worth it...